subsys/bluetooth/controller/hci/hci.c

```diff
@@ -5817,33 +5817,12 @@ int hci_iso_handle(struct net_buf *buf, struct net_buf **evt)
 			return -EINVAL;
 		}
 
-		/* FIXME: convey group start */
-		sdu_frag_tx.grp_ref_point = 0;
-
-		/* FIXME: temporary interface to enable ISOAL data Tx
-		 * Create provide proper interface between client
-		 * (using ISOAL target_event) and ISOAL, preferably
-		 * without dependence on peeking at LL data.
-		 * Problem is that client must specify a value greater
-		 * than LL bisPayloadCounter or no data is sent.
-		 */
 		lll_iso = &adv_iso->lll;
-
-		/* FIXME: Remove the below temporary hack to buffer up ISO data
-		 * if the SDU interval and ISO interval misalign.
-		 */
-		uint64_t pkt_seq_num = lll_iso->payload_count / lll_iso->bn;
-
-		if (((pkt_seq_num - stream->pkt_seq_num) & BIT64_MASK(39)) <=
-		    BIT64_MASK(38)) {
-			stream->pkt_seq_num = pkt_seq_num;
-		} else {
-			pkt_seq_num = stream->pkt_seq_num;
-		}
-
-		sdu_frag_tx.target_event = pkt_seq_num;
-
-		stream->pkt_seq_num++;
+		sdu_frag_tx.target_event = lll_iso->payload_count / lll_iso->bn;
```

Comment:
`lll_iso->payload_count`  is the value for the next BIG event, hence, the target_event is correct.

Reply:
Is that true even if the SDU Interval is less that the ISO interval? i.e. there are multiple SDU per ISO interval?

I guess I have two concerns which I am not sure are entirely valid:
-- Is there a possibility that this floor division yields an event that is already in progress?
-- In CIS we check ull_ref_get(&cig->ull) to decide if we are past the subevents for the current event. This is done to avoid a situation where the SDU is received too close to the start of the next event such that the fragmentation and propagation could result in payload being too late for transmission. Is this a concern for BIS as well?

Reply:
Thanks for explaining, i have added your comment in the code.